appender.go

```diff
@@ -236,169 +230,188 @@ func mallocLogicalTypeSlice(count int) (unsafe.Pointer, []C.duckdb_logical_type)
 	return ctPtr, slice
 }
 
-func mallocCStringSlice(count int) (unsafe.Pointer, []*C.char) {
-	var dummy *C.char
-	size := C.size_t(unsafe.Sizeof(dummy))
+func isNull[T any](val any) bool {
+	if reflect.ValueOf(val).Kind() != reflect.Ptr {
+		return false
+	}
+	v := val.(*T)
+	if v != nil {
+		return false
+	}
+	return true
+}
 
-	csPtr := unsafe.Pointer(C.malloc(C.size_t(count) * size))
-	slice := (*[1 << 30]*C.char)(csPtr)[:count:count]
-	return csPtr, slice
+func getValue[T any](val any) T {
+	if reflect.ValueOf(val).Kind() != reflect.Ptr {
+		return val.(T)
+	}
+	return *(val.(*T))
```

Comment:
There's a nil pointer dereference here, called from the `colInfo` generated in `initCString` (`blob` branch) and called from `setList`'s `childInfo.fn`

Reply:
Nice catch - I was missing an early out. On this note, we need to add more testing to these changes. We now allow passing pointers also for top-level types. I am already extending some test cases, but please drop anything here that you can find that still breaks!

Reply:
Our test suite (of various types and lists of those types) is now passing without issue, thanks for the rapid responses.